ops/attrs.rs

```diff
@@ -1,44 +1,61 @@
 // Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
 use syn::parse::Parse;
 use syn::parse::ParseStream;
-use syn::punctuated::Punctuated;
 use syn::Error;
 use syn::Ident;
 use syn::Result;
 use syn::Token;
 
-#[derive(Copy, Clone, Debug, Default)]
+#[derive(Clone, Debug, Default)]
 pub struct Attributes {
   pub is_unstable: bool,
   pub is_v8: bool,
   pub must_be_fast: bool,
   pub deferred: bool,
   pub is_wasm: bool,
+  pub relation: Option<Ident>,
 }
 
 impl Parse for Attributes {
   fn parse(input: ParseStream) -> Result<Self> {
-    let vars = Punctuated::<Ident, Token![,]>::parse_terminated(input)?;
-
-    let vars: Vec<_> = vars.iter().map(Ident::to_string).collect();
-    let vars: Vec<_> = vars.iter().map(String::as_str).collect();
-    for var in vars.iter() {
-      if !["unstable", "v8", "fast", "deferred", "wasm"].contains(var) {
-        return Err(Error::new(
-          input.span(),
-          "invalid attribute, expected one of: unstable, v8, fast, deferred, wasm",
-        ));
+    let mut self_ = Self::default();
+    let mut fast = false;
+    loop {
+      match input.parse::<Ident>() {
+        Ok(v) => {
+          match v.to_string().as_str() {
+            "unstable" => self_.is_unstable = true,
+            "v8" => self_.is_v8 = true,
+            "fast" => fast = true,
+            "deferred" => self_.deferred = true,
+            "wasm" => self_.is_wasm = true,
+            "slow" => {
+              if !fast {
+                return Err(Error::new(
+                  input.span(),
+                  "relational attributes can only be used with fast attribute",
+                ));
+              }
+              input.parse::<Token![=]>()?;
+              self_.relation = Some(input.parse()?);
+            }
+            _ => {
+              return Err(Error::new(
+             input.span(),
+            "invalid attribute, expected one of: unstable, v8, fast, deferred, wasm",
+            ));
+            }
+          };
+          let _ = input.parse::<Token![,]>();
+        }
+        Err(_) => {
+          break;
+        }
       }
     }
 
-    let is_wasm = vars.contains(&"wasm");
+    self_.must_be_fast = self_.is_wasm || fast;
```

Comment:
I guess this is still not being asserted anywhere?

Reply:
hmm, it is handled in the optimizer: https://github.com/denoland/deno/blob/888ceac7fdc6e9b96d860f183dd2f31df3f3d601/ops/optimizer.rs#L368-L369

doesn't look like `fast_compatible` is asserted though anywhere.